packages/ra-auth-cognito/src/MfaTotpAssociationForm.tsx

```diff
@@ -0,0 +1,97 @@
+import { Box, Button, CardContent, CircularProgress } from '@mui/material';
+import clsx from 'clsx';
+import { QRCodeSVG } from 'qrcode.react';
+import * as React from 'react';
+import {
+    Form,
+    LoginFormClasses,
+    TextInput,
+    required,
+    useTranslate,
+} from 'react-admin';
+import type { SubmitHandler } from 'react-hook-form';
+import type { FormData } from './useCognitoLogin';
+
+export type MfaTotpAssociationFormProps = {
+    className?: string;
+    submit: SubmitHandler<FormData>;
+    isLoading?: boolean;
+    secretCode: string;
+    username: string;
+    applicationName: string;
+};
+
+export const MfaTotpAssociationForm = ({
+    className,
+    submit,
+    isLoading,
+    secretCode,
+    username,
+    applicationName,
+}: MfaTotpAssociationFormProps) => {
+    const translate = useTranslate();
+    const otpUrl = `otpauth://totp/${username}?secret=${secretCode}&issuer=${applicationName}`;
+
+    return (
+        <Form
+            onSubmit={submit}
+            mode="onChange"
+            noValidate
+            className={clsx('RaLoginForm-root', className)}
+        >
+            <CardContent className={LoginFormClasses.content}>
+                <Box sx={{ maxWidth: '720px', pb: 2, margin: 'auto' }}>
+                    {translate('ra.auth.cognito.mfa.totp.association_required')}
```

Comment:
<!-- Nitpicks are trivial preference-based requests. These should be non-blocking by nature.  -->
**nitpick (non-blocking):** ‏It would be nice to add a default english string for all new translation keys

```suggestion
                    {translate('ra.auth.cognito.mfa.totp.association_required', { _: 'Two steps authentication is required, please associate an authenticator to your account by scanning the QR code or entering the secret key.' })}
```